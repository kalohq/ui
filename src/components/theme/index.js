/* @flow */
export {default} from './theme';
