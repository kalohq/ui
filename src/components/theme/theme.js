/* @flow */

import tokens from '../../design-tokens/tokens.theme.js';

/**
 * ThemeProvider Kalo Theme
 * 
 * This is the core CSS-in-JS theme that is passed in to the
 * global ThemeProvider. It is mainly composed of our core
 * design tokens.
 */

const theme = {
  space: [0, 8, 16, 32, 64],
  breakpoints: [],
  ...tokens,
};

export default theme;
