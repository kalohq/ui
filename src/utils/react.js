import ReactDOM from 'react-dom';

/**
 * Focus a ref’d react input when it is mounted
 * @param ref
 */
export function focusOnMount(ref) {
  const el = ReactDOM.findDOMNode(ref);

  if (el) {
    // Give focus and put the cursor at the end of
    // the value.
    el.focus();
    el.value = el.value;
  }
}

/**
 * Select the text of a ref’d react input when it is mounted
 * @param ref
 */
export function selectOnMount(ref) {
  if (ref && ref.select) {
    ref.select();
  }
}
