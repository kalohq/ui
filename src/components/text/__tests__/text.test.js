/* @flow */
/* eslint-env jest */
import * as React from 'react';
import {sheet} from 'emotion';
import serializer from 'jest-glamor-react';
import renderer from 'react-test-renderer';
import {ThemeProvider} from 'emotion-theming';

import theme from 'components/theme';
import Text from 'components/text';

expect.addSnapshotSerializer(serializer(sheet));

describe('Text', () => {
  const defaultProps = {
    size: 'medium',
    color: 'charcoal',
  };
  const create = (props = {}) =>
    renderer
      .create(
        <ThemeProvider theme={theme}>
          <Text {...defaultProps} {...props}>
            Hello World
          </Text>
        </ThemeProvider>
      )
      .toJSON();

  test('should render shallow component ok', () => {
    const element = create();
    expect(element).toMatchSnapshot();
  });

  test('should render green text', () => {
    const element = create({color: 'green'});
    expect(element).toMatchSnapshot();
  });

  test('should render large text', () => {
    const element = create({size: 'large'});
    expect(element).toMatchSnapshot();
  });
});
