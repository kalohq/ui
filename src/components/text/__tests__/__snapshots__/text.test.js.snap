// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text should render green text 1`] = `
.glamor-1 {
  display: inline-block;
  vertical-align: bottom;
  font-family: 'WebFaktSoftPro',sans-serif;
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
  cursor: inherit;
  text-align: unset;
  text-decoration: inherit;
  font-weight: 400;
  font-size: 16px;
  color: rgb(62,179,138);
}

.glamor-1:hover {
  text-decoration: inherit;
}

.glamor-1 strong {
  font-weight: 500;
}

<span
  className="glamor-0 glamor-1"
  color="green"
  onClick={undefined}
  size="medium"
  style={Object {}}
>
  Hello World
</span>
`;

exports[`Text should render large text 1`] = `
.glamor-1 {
  display: inline-block;
  vertical-align: bottom;
  font-family: 'WebFaktSoftPro',sans-serif;
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
  cursor: inherit;
  text-align: unset;
  text-decoration: inherit;
  font-weight: 400;
  font-size: 18px;
  color: rgb(55,69,97);
}

.glamor-1:hover {
  text-decoration: inherit;
}

.glamor-1 strong {
  font-weight: 500;
}

<span
  className="glamor-0 glamor-1"
  color="charcoal"
  onClick={undefined}
  size="large"
  style={Object {}}
>
  Hello World
</span>
`;

exports[`Text should render shallow component ok 1`] = `
.glamor-1 {
  display: inline-block;
  vertical-align: bottom;
  font-family: 'WebFaktSoftPro',sans-serif;
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
  cursor: inherit;
  text-align: unset;
  text-decoration: inherit;
  font-weight: 400;
  font-size: 16px;
  color: rgb(55,69,97);
}

.glamor-1:hover {
  text-decoration: inherit;
}

.glamor-1 strong {
  font-weight: 500;
}

<span
  className="glamor-0 glamor-1"
  color="charcoal"
  onClick={undefined}
  size="medium"
  style={Object {}}
>
  Hello World
</span>
`;
